drivers/spi/spi_xec_gpspi.c

```diff
@@ -0,0 +1,791 @@
+	}
+
+	regs->clock_control &= ~(XEC_GPSPI_CLK_CTRL_PH_MSK);
+	regs->clock_control |= gpspi_spi_mode_tbl[index & 0x3u];
+
+	return 0;
+}
+
+/* NOTE: struct spi_config.operation is defined as uint32_t when
+ * CONFIG_SPI_EXTENDED_MODES is enabled otherwise it is uint16_t.
+ * SPI_LINES_xxx are located at bit 16.
+ */
+static int gpspi_check_unsupported_features(const struct spi_config *spi_conf)
+{
+	uint32_t lines = spi_conf->operation & SPI_LINES_MASK;
+
+	if (spi_conf->operation & (SPI_OP_MODE_SLAVE | SPI_MODE_LOOP)) {
+		LOG_ERR("Does not support SPI device or loop back");
+		return -ENOTSUP;
+	}
+
+	if (IS_ENABLED(CONFIG_SPI_EXTENDED_MODES)) {
+		if (lines != SPI_LINES_SINGLE) {
+			LOG_ERR("Supports single(full-duplex) mode only");
+			return -ENOTSUP;
+		}
+	}
+
+	if (spi_conf->operation & SPI_CS_ACTIVE_HIGH) {
+		LOG_ERR("CS active high not supported");
+		return -ENOTSUP;
+	}
+
+	if (SPI_WORD_SIZE_GET(spi_conf->operation) != 8) {
+		LOG_ERR("Word size != 8 not supported");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int req_full_reconfig(const struct device *dev, const struct spi_config *spi_conf)
+{
+	struct spi_xec_gpspi_data * const data = dev->data;
+
+	if (data->scfg.frequency != spi_conf->frequency) {
+		return 1; /* frequency change */
+	} else if ((data->scfg.operation & SPI_MODE_MASK)
+		   != (spi_conf->operation & SPI_MODE_MASK)) {
+		return 1; /* CPOL and/or CPHA changed */
+	}
+
+	return 0;
+}
+
+/* Configure GPSPI for full-duplex operation.
+ * TODO - configuration will become complex if driver supports CONFIG_SPI_EXTENDED_MODES
+ * If target device is a SPI flash the commands for dual/quad involve multiple phases:
+ * transmit command, transmit address, optional transmit mode byte, optional transmit
+ * clocks with I/O tri-stated, read data using multiple I/O pins.
+ * The SPI driver specifies the I/O mode in struct spi_config on each call to the
+ * transfer API. This means struct spi_config contents can change on every call and
+ * the previous call may have left the SPI open (chip select asserted). For this case
+ * reconfiguring the controller should not cause any SPI signals to glitch.
+ * Rules:
+ * If GPSPI.control.CE is 1 then chip select is asserted
+ *   Ignore frequency change. Should we return error if new frequency?
+ *   Record other flags such as SPI_HOLD_ON_CS and SPI_LOCK_ON
+ *   Should we allow MSBF/LSBF change?
+ *   Allow I/O mode change: half-duplex, full-duplex, and dual. Reject quad.
+ * Else
+ *   Allowed to do full reconfigure (controller reset)
+ * Endif
+ *
+ */
+static int gpspi_configure(const struct device *dev, const struct spi_config *spi_conf)
+{
+	const struct spi_xec_gpspi_config * const cfg = dev->config;
+	struct spi_xec_gpspi_data * const data = dev->data;
+	struct xec_gpspi_regs * const regs = cfg->regs;
+	struct spi_context *ctx = &data->ctx;
+	int ret;
+	uint8_t ctrl;
+
+	/* Need to reconfigure? */
+	ret = memcmp(&data->scfg, spi_conf, sizeof(struct spi_config));
```

Comment:
just call spi_context_configured()

Reply:
Inserted the spi_context_configured api to verify already configured or not.